texar/data/data/multi_aligned_data.py

```diff
@@ -475,166 +484,43 @@ def process(self, raw_example: Tuple[Union[str, Dict[str, Any]], ...]) \
         batch: Dict[str, Any] = {}
         for i, transposed_example in enumerate(transposed_examples):
             if _is_text_data(dataset_hparams[i]["data_type"]):
-                # Treat it as MonoTextData and collate
-                pad_length = dataset_hparams[i]["max_seq_length"]
-                if pad_length is not None:
-                    pad_length += sum(x is not None and x != '' for x in
-                                      [self.vocab(i).bos_token,
-                                       self.vocab(i).eos_token])
-                text_ids = [self.vocab(i).map_tokens_to_ids_py(sent) for sent in
-                            transposed_example]
-                text_ids, lengths = padded_batch(
-                    text_ids, pad_length, pad_value=self.vocab(i).pad_token_id)
-                # also pad the examples
-                pad_length = pad_length or max(lengths)
-                transposed_example = tuple(
-                    sent + [''] * (pad_length - len(sent))
-                    if len(sent) < pad_length else sent
-                    for sent in transposed_example
-                )
-
-                text_ids = torch.from_numpy(text_ids).to(device=self.device)
-                lengths = torch.tensor(lengths, dtype=torch.long,
-                                       device=self.device)
-
-                batch.update(
-                    {connect_name(self._name_prefix[i], "text"):
-                     transposed_example,
-                     connect_name(self._name_prefix[i], "text_ids"):
-                     text_ids,
-                     connect_name(self._name_prefix[i], "length"):
-                     lengths})
+                text_batch = self._databases[i].collate(transposed_example)
+                batch.update(text_batch._batch)
             elif _is_scalar_data(dataset_hparams[i]["data_type"]):
-                # convert the list of strings into appropriate tensors here
-                data_type = dataset_hparams[i]["data_type"]
-                if data_type == "int":
-                    example_np = np.array(transposed_example, dtype=np.int32)
-                    example = torch.from_numpy(example_np). \
-                        to(device=self.device)
-                elif data_type == "float":
-                    example_np = np.array(transposed_example, dtype=np.float32)
-                    example = torch.from_numpy(example_np). \
-                        to(device=self.device)
-                else:
-                    raise ValueError(
-                        "Incorrect \'data_type\'. Currently \'int\' and "
-                        "\'float\' are supported. Received {}"
-                        .format(data_type))
-                batch.update({self._name_prefix[i]: example})
+                scalar_batch = self._databases[i].collate(transposed_example)
+                batch.update(scalar_batch._batch)
             elif _is_record_data(dataset_hparams[i]["data_type"]):
```

Comment:
Same here, redundant checks.

Reply:
Resolved in 462312cab452a7f1c3eacce634d65f1a50c2fbbc